### PR TITLE
Simplify the calico wrapper

### PR DESCRIPTION
**Pull request by janisz**
* Do not download CA cert – use the one provided by dcos-cli
* Check https for enterprise – this will test if cert is loaded
* Remove unused calico client

Fixes: https://jira.d2iq.com/browse/COPS-6353

### CODE REVIEW

File: pkg/cmd/calico/calico.go [hunk 1]
```diff
@@ -99,50 +97,44 @@ func getEnvironment(ctx api.Context, grpcPort string) ([]string, error) {
...
-	caClient := newClient(httpClient)
-	caCert, err := caClient.getCaCertificate()
-	if err != nil {
-		return nil, fmt.Errorf("could not get certificate: %s", err)
+	tlsCAPath, ok := os.LookupEnv("DCOS_TLS_CA_PATH")
+	if !ok {
+		return nil, fmt.Errorf("DCOS_TLS_CA_PATH is not defined. " +
+			"Calico requires secure connection. " +
+			"Do NOT use --insecure flag with dcos cluster setup, " +
+			"--no-check should be used instead")
```
Comment (pierrebeitz) [inline pkg/cmd/calico/calico.go:120]: nice 2 know!

File: pkg/cmd/calico/calico.go [hunk 2]
```diff
@@ -99,50 +97,44 @@ func getEnvironment(ctx api.Context, grpcPort string) ([]string, error) {
...
 			fmt.Sprintf("ETCD_CUSTOM_GRPC_METADATA=authorization:token=%s", cluster.ACSToken()),
 			fmt.Sprintf("ETCD_ENDPOINTS=%s%s", host, grpcPort),
 		}, nil
 	}
 
-	caClient := newClient(httpClient)
-	caCert, err := caClient.getCaCertificate()
-	if err != nil {
-		return nil, fmt.Errorf("could not get certificate: %s", err)
+	tlsCAPath, ok := os.LookupEnv("DCOS_TLS_CA_PATH")
```
Comment (pierrebeitz) [inline pkg/cmd/calico/calico.go:115]: formerly we requested it via `/ca/dcos-ca.crt`, right? 
why can we change to this method without affecting operators?

Comment (janisz) [reply pkg/cmd/calico/calico.go:115]: > formerly we requested it via /ca/dcos-ca.crt, right?

Yes and we download this cert every time duplication job that is done during `dcos cluster setup`. [dcos-cli sets this variable](https://github.com/dcos/dcos-cli/blob/b8f810a03d4e989ec02bb98d6685e4eff2d7ceb5/design/plugin.md) for us so we do not need to download cert every time user call `dcos calico`.  

>- `DCOS_TLS_CA_PATH`: Unless `DCOS_TLS_INSECURE=1`, this env var indicates the path to the CA bundle to
                      verify server certificates against. When it is not set, the system's CA bundle must be used.

I mentioned it before https://github.com/dcos/dcos-core-cli/pull/427#discussion_r390294992 but I have no time to properly debug the problem.

File: pkg/cmd/calico/calico.go (hunk 1 above)
Comment (janisz) [reply pkg/cmd/calico/calico.go:120]: Refs: https://github.com/dcos/dcos-cli/pull/1535/files

File: pkg/cmd/calico/calico.go (hunk 2 above)
Comment (pierrebeitz) [reply pkg/cmd/calico/calico.go:115]: i love this cleanup!

**Review by pierrebeitz — APPROVED** [review]

**Pull request merged**